libs/libjson/CMakeLists.txt

```diff
@@ -1,38 +1,5 @@
 # ADDING JSON
 
-set ( JSON_HEADERS
```

Comment:
I think is important that all headers are added to the library as well, not only sources. 
Many IDE's like qt creator require that in order to include them in the project. 
If you find painful adding the files by hand you can do a cmake file glob